bears/general/HTTPSBear.py

```diff
@@ -0,0 +1,72 @@
+from urllib.parse import urlparse
+
+from bears.general.URLBear import URLBear
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+from coalib.results.Diff import Diff
+from coalib.settings.Setting import typed_dict
+
+
+class HTTPSBear(LocalBear):
+    DEFAULT_TIMEOUT = 15
+    LANGUAGES = {'All'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Documentation'}
+    BEAR_DEPS = {URLBear}
+
+    def run(self, filename, file, dependency_results=dict(),
+            network_timeout: typed_dict(str, int, DEFAULT_TIMEOUT)=dict()):
+        """
+        Find http links in any text file and check if the https version of
+        link is valid. If so, provide option for replacing it with https.
+
+        An https link is considered valid if the server responds with a 2xx
+        code.
+
+        Warning: This bear will make HEAD requests to all URLs mentioned in
+        your codebase, which can potentially be destructive. As an example,
+        this bear would naively just visit the URL from a line that goes like
+        `do_not_ever_open = 'https://api.acme.inc/delete-all-data'` wiping out
+        all your data.
+
+        :param dependency_results:
+            Results given by URLBear.
+        """
+        for result in dependency_results.get(URLBear.name, []):
+            line_number, link, code, context = result.contents
+            if (link[0:5] != 'https'):
+                httpslink = 'https' + link[4:]
```

Comment:
to avoid maintenance problems later, I would advise to put the `https` string into a constant, e.g. `HTTPS_PREFIX`. You've also hardcoded the number `4` now that comes from the length of `http`, so maybe we also want a `HTTP_PREFIX`. So in the end having something like this avoids problems:
```
httpslink = HTTPS_PREFIX + link[len(HTTP_PREFIX):]
```
what do you think? ;)